drivers/soundwire/stream.c

```diff
@@ -1950,35 +1953,39 @@ int sdw_stream_add_slave(struct sdw_slave *slave,
 			"alloc master runtime failed for stream:%s\n",
 			stream->name);
 		ret = -ENOMEM;
-		goto error;
+		goto unlock;
 	}
-	ret =  sdw_master_rt_config(m_rt, stream_config);
 
 skip_alloc_master_rt:
 	s_rt = sdw_slave_rt_alloc(slave, m_rt);
 	if (!s_rt) {
 		dev_err(&slave->dev,
 			"Slave runtime alloc failed for stream:%s\n",
 			stream->name);
+		alloc_slave_rt = false;
 		ret = -ENOMEM;
-		goto stream_error;
+		goto alloc_error;
 	}
 
-	ret = sdw_slave_rt_config(s_rt, stream_config);
+	ret = sdw_slave_port_alloc(slave, s_rt, num_ports);
 	if (ret)
-		goto stream_error;
+		goto alloc_error;
 
-	ret = sdw_config_stream(&slave->dev, stream, stream_config, true);
+	ret =  sdw_master_rt_config(m_rt, stream_config);
 	if (ret)
-		goto stream_error;
+		goto unlock;
```

Comment:
We allocate slave rt successfully already. So I think we should `goto alloc_error` and same as all the `goto unlock` below.

Reply:
will revisit, thanks!

Reply:
same, this was intentional: we only return an errror code on a config error.